### PR TITLE
feat: allow status.progress_bar to be used as a context manager

### PR DESCRIPTION
**Pull request by seantur**
Hi! Thanks for all the good work with marimo. I found myself wanting to update the subtitle while the progress bar was running, which is a pattern I've used before with `tqdm`.

I modified progress_bar to also be usable as a context manager, which exposes the underlying `ProgressBar` and allows for manual calls to `update`. This achieves the functionality I'm looking for, and still allows for using the normal iterable method.

Happy to make any changes if needed 😄

### CODE REVIEW

**Comment by seantur** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by akshayka — DISMISSED** [review]
Thanks! Nice API. This is definitely an improvement.

File: marimo/_plugins/stateless/status/_progress.py [hunk 1]
```diff
@@ -295,43 +285,91 @@ def progress_bar(
...
+    the bar:
+
+    ```python
+    with mo.status.progress_bar(total=10) as bar:
+        for i in range(10):
+            ...
+            bar.update()
+    ```
+
+The `update` method accepts the optional keyword arguments `increment` (defaults to `1`), `title`, and `subtitle`.
```
Comment (akshayka) [inline marimo/_plugins/stateless/status/_progress.py:298]:
```suggestion
The `update` method accepts the optional keyword
arguments `increment` (defaults to `1`), `title`, and `subtitle`.
```

File: marimo/_plugins/stateless/status/_progress.py [hunk 2]
```diff
@@ -295,43 +285,92 @@ def progress_bar(
...
+
+    ```python
+    with mo.status.progress_bar(total=10) as bar:
+        for i in range(10):
+            ...
+            bar.update()
+    ```
+
+The `update` method accepts the optional keyword
+arguments `increment` (defaults to `1`), `title`, and `subtitle`.
```
Comment (akshayka) [inline marimo/_plugins/stateless/status/_progress.py:299]:
```suggestion
    The `update` method accepts the optional keyword
    arguments `increment` (defaults to `1`), `title`,
    and `subtitle`.
    
```

**Pull request merged**